test/src/monitoring_frame_serialization.cpp

```diff
@@ -58,4 +65,26 @@ DynamicSizeRawData serialize(MonitoringFrameMsg& frame)
 
   return raw_processing::toArray<DynamicSizeRawData>(os);
 }
-}  // namespace psen_scan_v2
\ No newline at end of file
+
+std::array<uint8_t, DIAGNOSTIC_DATA_LENGTH_IN_BYTES>
+serializeDiagnosticMessages(std::vector<MonitoringFrameDiagnosticMessage>& messages)
+{
+  std::array<uint8_t, DIAGNOSTIC_DATA_LENGTH_IN_BYTES> raw_diagnostic_data;
+  raw_diagnostic_data.fill(0);
+
+  for (const auto& elem : messages)
+  {
+    raw_diagnostic_data.at(DIAGNOSTIC_MESSAGE_UNUSED_OFFSET_IN_BYTES +
```

Comment:
Its not really understandable how the index is calculated. Maybe a sub-function would help?

Reply:
On it